webapi/CopilotChat/Controllers/ServiceOptionsController.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using SemanticKernel.Service.CopilotChat.Models;
+using SemanticKernel.Service.Options;
+
+namespace SemanticKernel.Service.CopilotChat.Controllers;
+
+/// <summary>
+/// Controller responsible for returning the service options to the client.
+/// </summary>
+[ApiController]
+[Authorize]
+public class ServiceOptionsController : ControllerBase
+{
+    private readonly ILogger<ServiceOptionsController> _logger;
+
+    private readonly MemoriesStoreOptions _memoriesStoreOptions;
+
+    public ServiceOptionsController(
+        ILogger<ServiceOptionsController> logger,
+        IOptions<MemoriesStoreOptions> memoriesStoreOptions)
+    {
+        this._logger = logger;
+        this._memoriesStoreOptions = memoriesStoreOptions.Value;
+    }
+
+    /// <summary>
+    /// Return the memory store type that is configured.
+    /// </summary>
+    [Route("memoriesStoreType")]
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public IActionResult GetMemoriesStoreType()
```

Comment:
Do you foresee us having to expose any other options from the backend (i.e., AI service, model used, etc.)? 

If so, this might make more sense as a GetServiceOptions call that returns a dictionary of KV pairings so we don't have to implement a getter for each Service Option we may need in the future. 

Also not blocking for this PR, we can change this in the future.

Reply:
I do foresee that, but we don't have a plan yet as what we need to include. And making this complicated will leave little room for design choices down the road unless we reimplement everything. I think we should keep this simple now so that it won't cause us much either way (i.e. one API per option v.s. one API for all options).

Reply:
Yeah sounds good, can you create an issue to track this? and link the number here? So others know not to keep adding getters for each option type

Reply:
+1 on using one API, instead of calling each API one-by-one when the app loads. 
I don't think the solution needs to be complicated and could be along the lines of:
```csharp
public IActionResult GetServiceOptions()
{
    return this.Ok(new
    {
        memoryStore = new
        {
            Types = Enum.GetNames(typeof(MemoriesStoreOptions.MemoriesStoreType)),
            SelectedType = this._memoriesStoreOptions.Type.ToString()
        }
    });
}
```


 it would be nice to go ahead and set apattern now, so that later, like you mentioned @TaoChenOSU, we don't have to reimplement things down the line, but I also don't want to block the PR more than I already have 😄 

if we don't end up doing it in this PR, can we instead just add a `// Note:` instead of creating an entire issue, since it won't currently be actionable?

Reply:
synced offline - we'll update this set a pattern and send a bag of options.